### PR TITLE
lint(engine): update lint scripts and config

### PR DESCRIPTION
**Pull request by gozssky**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tiflow/issues/7475

### What is changed and how it works?

* Add depguard config to check errors import path for engine.
* Fix incorrect `--skip-dirs` in golangci-lint script.
* Fix several log import paths.

#### Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

### Release note 

```release-note
None
```

### CODE REVIEW

File: engine/jobmaster/dm/metadata/unit_state.go [hunk 1]
```diff
@@ -28,13 +28,14 @@ import (
 
 // UnitState represents the state of units.
 type UnitState struct {
-	state
+	state //nolint:unused
```
Comment (lance6716) [inline engine/jobmaster/dm/metadata/unit_state.go:31]: I forget why we must embed it into a customized State. Can we remove it now? @GMHDBJD

Comment (gozssky) [reply engine/jobmaster/dm/metadata/unit_state.go:31]: `state` is an empty interface. From my understanding, it is not much different from a raw interface.

**Review by amyangfei — APPROVED** [review]

File: .golangci.yml [hunk 2]
```diff
@@ -60,6 +60,18 @@ linters-settings:
...
+        include-go-root: true
+        packages:
+          - errors
+          - github.com/pingcap/errors
+        packages-with-error-message:
+          - errors: "error handling is allowed only by github.com/pingcap/tiflow/pkg/errors"
+          - github.com/pingcap/errors: "error handling is allowed only by github.com/pingcap/tiflow/pkg/errors"
+        ignore-file-rules:
+          - "!**/tiflow/engine/**"
+          - "**/engine/pkg/rpcutil/*.go"
```
Comment (amyangfei) [inline .golangci.yml:74]: This rule is confused, I wonder why `!**/tiflow/cdc/**` is not configured, and it is not checked.

Comment (gozssky) [reply .golangci.yml:74]: 1. `"!**/tiflow/engine/**"` means all files except `engine` directory should be excluded.
2. `"**/engine/pkg/rpcutil/*.go"` means `engine/pkg/rpcutil` directory should be excluded.

`tiflow/cdc` is excluded in rule 1.

**Review by lance6716 — APPROVED** [review]

File: engine/jobmaster/dm/metadata/unit_state.go (hunk 1 above)
Comment (GMHDBJD) [reply engine/jobmaster/dm/metadata/unit_state.go:31]: We can remove it.

Comment (gozssky) [reply engine/jobmaster/dm/metadata/unit_state.go]: Removed in https://github.com/pingcap/tiflow/pull/7565/commits/ff2ac803e5b92130b448c43c47a29b9b10f3ddb4. @lance6716 @GMHDBJD PTAL again.

**Review by lance6716 — COMMENTED** [review]
lgtm

**Pull request merged**